LanguageFeatures/Extension-types/exhaustiveness_map_A01_t01.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with map as a matched type can
+/// never be exhaustive
+///
+/// @description Check that a map pattern cannot be exhaustive. Test an
+/// extension type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type ET1<K, V>(Map<K, V> _) {}
+extension type ET2<K, V>(Map<K, V> _) implements Map<K, V> {}
+
+String test1_1(ET1<bool, bool> m) =>
+  switch (m) { // Empty map is not matched
```

Comment:
We're also not matching a map with two key/value pairs. There would be 4 of those.

But it doesn't matter much whether they are added or not, because we can't even express the empty map as a map pattern, so we cannot exhaust the conceptual/intuitive set of maps of type `Map<bool, bool>` using map patterns.

However, in this context it could be useful to add an extra test: Start with the same patterns, just so we have something for the exhaustiveness algorithm to work with, and then end with an exhaustive pattern like `Map()`.

This will give us exhaustiveness for the switch expression _if_ the matched value type is treated as its representation type. (If it is treated as the extension type `ET1` then there is no reason to think that `Map()` will exhaust the matched value type). So that would be a quite useful test.